wdl/MainVcfQc.wdl

```diff
@@ -0,0 +1,843 @@
+    preemptible: select_first([runtime_override.preemptible_tries, runtime_default.preemptible_tries])
+    maxRetries: select_first([runtime_override.max_retries, runtime_default.max_retries])
+    docker: sv_pipeline_qc_docker
+    bootDiskSizeGb: select_first([runtime_override.boot_disk_gb, runtime_default.boot_disk_gb])
+  }
+
+  command <<<  
+    set -eu -o pipefail
+      
+    # Plot benchmarking stats
+    /opt/sv-pipeline/scripts/vcf_qc/plotQC.external_benchmarking.helper.sh \
+      ~{benchmarking_tarball} \
+      ~{comparator}
+
+    # Prep outputs
+    tar -czvf ~{prefix}.collectQC_benchmarking_~{comparator}_output.wPlots.tar.gz \
+    collectQC_benchmarking_~{comparator}_output
+  >>>
+
+  output {
+    File tarball_wPlots = "~{prefix}.collectQC_benchmarking_~{comparator}_output.wPlots.tar.gz"
+  }
+}
+
+
+# Plot per-sample stats
+task PlotQcPerSample {
+  input {
+    File vcf_stats
+    File samples_list
+    File per_sample_tarball
+    String prefix
+    String sv_pipeline_qc_docker
+    RuntimeAttr? runtime_attr_override
+  }
+  RuntimeAttr runtime_default = object {
+    mem_gb: 7.75,
+    disk_gb: 50,
+    cpu_cores: 1,
+    preemptible_tries: 1,
+    max_retries: 1,
+    boot_disk_gb: 10
+  }
+
+  RuntimeAttr runtime_override = select_first([runtime_attr_override, runtime_default])
+  runtime {
+    memory: "~{select_first([runtime_override.mem_gb, runtime_default.mem_gb])} GiB"
+    disks: "local-disk ~{select_first([runtime_override.disk_gb, runtime_default.disk_gb])} HDD"
+    cpu: select_first([runtime_override.cpu_cores, runtime_default.cpu_cores])
+    preemptible: select_first([runtime_override.preemptible_tries, runtime_default.preemptible_tries])
+    maxRetries: select_first([runtime_override.max_retries, runtime_default.max_retries])
+    docker: sv_pipeline_qc_docker
+    bootDiskSizeGb: select_first([runtime_override.boot_disk_gb, runtime_default.boot_disk_gb])
+  }
+
+  command <<<
+    set -eu -o pipefail
+    
+    # Make per-sample directory
+    mkdir ~{prefix}_perSample/
+
+    # Untar per-sample VID lists
+    mkdir tmp_untar/
+    tar -xvzf ~{per_sample_tarball} \
+      --directory tmp_untar/
+    find tmp_untar/ -name "*.VIDs_genotypes.txt.gz" | while read FILE; do
+      mv $FILE ~{prefix}_perSample/
+    done
+
+    # Plot per-sample distributions
+    /opt/sv-pipeline/scripts/vcf_qc/plot_sv_perSample_distribs.R \
+      -S /opt/sv-pipeline/scripts/vcf_qc/SV_colors.txt \
+      ~{vcf_stats} \
+      ~{samples_list} \
+      ~{prefix}_perSample/ \
+      ~{prefix}_perSample_plots/
+
+    # Prepare output
+    tar -czvf ~{prefix}.plotQC_perSample.tar.gz \
+      ~{prefix}_perSample_plots
+  >>>
+
+  output {
+    File perSample_plots_tarball = "~{prefix}.plotQC_perSample.tar.gz"
+  }
+}
+
+
+# Plot per-family stats
+task PlotQcPerFamily {
+  input {
+    File vcf_stats
+    File samples_list
+    File ped_file
+    File per_sample_tarball
+    Int max_trios
+    Int? random_seed = 2021
```

Comment:
We have gotten feedback that collaborators using MiniWDL have encountered issues with this method of setting defaults. To work around this, it's safer to leave this line as `Int? random_seed` and then within the task before the command block do the following and then use `random_seed_` within the command block:
```
Int random_seed_ = if (defined(random_seed)) then random_seed else 2021
```